### ISSUE TITLE
User cannot use just deleted role name again

### ISSUE THREAD

**Issue by plakitkelly** [keeleinstituut/tv-tolkevarav#407]
Steps:
1. Create new role
2. Name it "test"
3. Delete this role
4. Create new role
5. Name it "test"

Actual: On the first time the role saved, on the second time user cannot see message "Roll x on edukalt loodud..". After refresh this role is gone.

Expected: If the user has deleted role x, he should be able to recreate a role with the same name in the future. 
In example, user has deleted role x. In the future he want to create a role with the same name. He has selected many privileges. First he sees that the role has been created. Then he goes to "Kasutajate haldus" and wants to assign this role to any user. He sees that this role is not in the list of roles. He didn't get any notification that the role has not been created before.
Link to video: [https://drive.google.com/file/d/1g-qjJI53yi4RNURR45YM9Kcu12SNSm1Q/view](https://drive.google.com/file/d/1g-qjJI53yi4RNURR45YM9Kcu12SNSm1Q/view)

If I haven't refreshed the page after creating the new role with same name and I try to delete this role, I see error message:

> The route api/roles/temp-cd4b5f6c-7a3e-4a4d-aa75-970d79723a03 could not be found

Link to video: [https://drive.google.com/file/d/1cdNp-FbtK5dUszwl1yK2EcniKDMrDs0v/view](https://drive.google.com/file/d/1cdNp-FbtK5dUszwl1yK2EcniKDMrDs0v/view)

**Comment by DanBondarenko** [comment]
@KaarelKa The `temp-` prefix in the UUID makes me suspect that this is a FE issue. Can you confirm?

**Comment by KaarelKa** [comment]
@DanBondarenko Yeah the delete issue seems to be FE related, however the issue with not being able to add a role with the same name after it has been deleted still seems to be a BE issue. Looks like the same issue we discussed in May https://pundar.slack.com/archives/C053FD55ATT/p1684826029819699 

Can basically already test it by trying to add the "Test" role. It is not returned from BE, when I fetch the list of all roles, however I also can't add it, since BE gives me the "The name has already been taken" error from POST https://api.dev.tolkevarav.eki.ee/authorization/api/roles endpoint.

**Comment by MariusJulius** [comment]
BE task: https://github.com/keeleinstituut/tv-tolkevarav/issues/432

**Comment by plakitkelly** [comment]
Testitud prelives. Korras.
